src/components/text-editor/text-editor.tsx

```diff
@@ -55,22 +99,51 @@ export class TextEditor {
             marks: schema.spec.marks,
         });
 
+        const plugins = this.getPlugins(mySchema);
+
         this.view = new EditorView(
             this.host.shadowRoot.querySelector('#editor'),
             {
                 state: EditorState.create({
                     doc: DOMParser.fromSchema(mySchema).parse(
                         this.host.shadowRoot.querySelector('#editor'),
                     ),
-                    plugins: exampleSetup({ schema: mySchema }),
+                    plugins: plugins,
                 }),
                 dispatchTransaction: (transaction) => {
                     const newState = this.view.state.apply(transaction);
                     this.view.updateState(newState);
 
-                    this.change.emit({ html: this.view.dom.innerHTML });
+                    this.change.emit({
+                        html: this.getHTML(),
+                    });
                 },
             },
         );
+
+        if (this.value) {
+            this.view.dom.innerHTML = this.value.html;
+        }
     }
+
+    private getHTML = (): string => {
+        if (
+            this.view.dom.textContent === '' ||
+            (this.view.dom.textContent === this.placeholder &&
+                this.view.dom.innerHTML !== `<p>${this.placeholder}</p>`) // TODO: this is a little too coupled to the placeholder plugin
```

Comment:
So, there is a plugin for adding a placeholder, but we still have to do stuff like this and `this.view.dispatch( this.view.state.tr.setMeta(this.placeholderKey, this.placeholder) )`?

I think this is way too much bending-over-backwards to support a placeholder. If we can't do it in a neater way, I think we should skip it.

Reply:
I think we should always do a dispatch since I think it's the only way we can communicate state changes from the web component to the editor (plugin) state 🤔.

But I also don't really like the `textContent` and `innerHTML` logic here. The last condition avoids emitting an empty string value when the value happens to be the same as the placeholder. I think there is probably a better way to avoid coupling the plugin to the generic transaction handler.